write-fonts/src/tables/variations.rs

```diff
@@ -370,9 +370,82 @@ impl Tuple {
     }
 }
 
+impl DeltaSetIndexMap {
+    /// Return the most compact entry format that can represent this mapping.
+    ///
+    /// EntryFormat is a packed u8 field that describes the compressed representation
+    /// of delta-set indices. For more info, see:
+    /// <https://learn.microsoft.com/en-us/typography/opentype/spec/otvarcommonformats#associating-target-items-to-variation-data>
+    // This is a direct port from fonttools' DeltaSetMap.getEntryFormat:
+    // https://github.com/fonttools/fonttools/blob/6d531f/Lib/fontTools/ttLib/tables/otTables.py#L644-L666
+    fn get_entry_format(mapping: &[u32]) -> EntryFormat {
+        let ored = mapping.iter().fold(0, |acc, idx| acc | *idx);
+
+        let inner = (ored & 0xFFFF) as u16;
+        let inner_bits = (16 - inner.leading_zeros() as u8).max(1);
```

Comment:
could we use [`u16::count_ones`](http://doc.rust-lang.org/1.73.0/std/primitive.u16.html#method.count_ones)?

Reply:
no we can't use count_ones here, that one simply returns the number of ones in the binary representation of self, whereas here we want to count the maximum number of bits that are sufficient to represent this u16.

```rust
>> let inner: u16 = 0b00000000_00010010;
>> inner.count_ones()
2
>> (16 - inner.leading_zeros() as u8).max(1)
5